protos/protos/public/modeldb/versioning/VersioningService.proto

```diff
@@ -418,6 +418,44 @@ message RevertRepositoryCommitsRequest {
     }
 }
 
+message FindRepositories {
+    repeated uint64 repo_ids = 1;
+    repeated string repo_names = 2;
+    string workspace_name = 3;
+    repeated KeyValueQuery predicates = 4;
+    bool ids_only = 5; // whether to return full experiment runs or ids only
+
+    //For pagination
+    int32 page_number = 6;
+    int32 page_limit = 7;
+    bool ascending = 8;
+    string sort_key = 9;
+
+    message Response {
+        repeated Repository repository = 1;
+        int64 total_records = 2;
+    }
+}
+
+message FindRepositoriesBlobs {
+    repeated uint64 repo_ids = 1;
+    repeated string repo_names = 2;
+    string workspace_name = 3;
+    repeated string location_prefix = 4;
+    int32 blob_type = 5;
+
+    //For pagination
+    int32 page_number = 6;
+    int32 page_limit = 7;
+    bool ascending = 8;
+    string sort_key = 9;
```

Comment:
if you end up not supporting this in first pass (since we do not know the exact requirement for sorting) comment these out. default sorting on updation time or creation time.

Reply:
make sense

Reply:
commentted